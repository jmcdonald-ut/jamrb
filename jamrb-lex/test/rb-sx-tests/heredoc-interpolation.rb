alive = 21
dying = 25

<<SIMPLE
#{alive} and well, full of optimism.
SIMPLE

<<"DOUBLEQUOTES"
#{dying} and well, dead.
DOUBLEQUOTES

<<'SINGLEHEREDOC'
Not escaped, interpolated, so on...#{alive} is simply alive.
SINGLEHEREDOC

<<`BACKTICK`
cd ~/projects && echo "I am #{dying}"
BACKTICK
