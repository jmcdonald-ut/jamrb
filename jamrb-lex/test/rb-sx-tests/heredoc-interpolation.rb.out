((Token ((1 0) on_ident "alive"))
 (Token ((1 5) on_sp " "))
 (Token ((1 6) on_op "="))
 (Token ((1 7) on_sp " "))
 (Token ((1 8) on_int "21"))
 (Token ((1 10) on_nl "\n"))
 (Token ((2 0) on_ident "dying"))
 (Token ((2 5) on_sp " "))
 (Token ((2 6) on_op "="))
 (Token ((2 7) on_sp " "))
 (Token ((2 8) on_int "25"))
 (Token ((2 10) on_nl "\n"))
 (Token ((3 0) on_ignored_nl "\n"))
 (Token ((4 0) on_heredoc_beg "<<SIMPLE"))
 (Token ((4 8) on_nl "\n"))
 (Token ((5 0) on_embexpr_beg "\\#{"))
 (Token ((5 2) on_ident "alive"))
 (Token ((5 7) on_embexpr_end "}"))
 (Token ((5 8) on_tstring_content " and well, full of optimism.\n"))
 (Token ((6 0) on_heredoc_end "SIMPLE\n"))
 (Token ((7 0) on_ignored_nl "\n"))
 (Token ((8 0) on_heredoc_beg "<<\"DOUBLEQUOTES\""))
 (Token ((8 16) on_nl "\n"))
 (Token ((9 0) on_embexpr_beg "\\#{"))
 (Token ((9 2) on_ident "dying"))
 (Token ((9 7) on_embexpr_end "}"))
 (Token ((9 8) on_tstring_content " and well, dead.\n"))
 (Token ((10 0) on_heredoc_end "DOUBLEQUOTES\n"))
 (Token ((11 0) on_ignored_nl "\n"))
 (Token ((12 0) on_heredoc_beg "<<'SINGLEHEREDOC'"))
 (Token ((12 17) on_nl "\n"))
 (Token
  ((13 0)
   on_tstring_content
   "Not escaped, interpolated, so on...\\#{alive} is simply alive.\n"))
 (Token ((14 0) on_heredoc_end "SINGLEHEREDOC\n"))
 (Token ((15 0) on_ignored_nl "\n"))
 (Token ((16 0) on_heredoc_beg "<<`BACKTICK`"))
 (Token ((16 12) on_nl "\n"))
 (Token ((17 0) on_tstring_content "cd ~/projects && echo \"I am "))
 (Token ((17 28) on_embexpr_beg "\\#{"))
 (Token ((17 30) on_ident "dying"))
 (Token ((17 35) on_embexpr_end "}"))
 (Token ((17 36) on_tstring_content "\"\n"))
 (Token ((18 0) on_heredoc_end "BACKTICK\n")))
